Mfa docs (#619)
tkhq/docs: authentication/mfa/enforcement-and-recovery.mdx

---
title: "MFA Enforcement and Recovery"
description: "Learn how to enforce multi-factor authentication (MFA) policies for end-users in your sub-organizations, and how to set up recovery mechanisms in case users lose access to their authentication methods."
sidebarTitle: "Enforcement and Recovery"
---

## Enforcing MFA for end users

> Enforcing MFA for an end-user on a sub-organization can do so using a [delegated access user](/concepts/policies/delegated-access-overview). 

Reviewer: ```suggestion
Enforcing MFA for an end user on a sub-organization can be done using a [delegated access user](/concepts/policies/delegated-access-overview). 
```

A delegated access user is a non-root user created in the sub-organization whose API key is controlled by the parent organization and has carefully scoped permissions to perform only specific actions.

To set this up:

1. The sub-organization's root user creates a delegated access user with an API key controlled by the parent org.
2. The sub-organization's root user creates a policy that allows the delegated access user to manage MFA policies.
3. The delegated access user can then create MFA policies for the sub-organization's root user.

The policy assigned to the delegated access user should be scoped to only allow MFA policy management:

``` ts
// Policy condition: only allow MFA policy activities
activity.resource == 'MFA_POLICY'
```

Once this is in place, the delegated access user (controlled by the parent org) can create MFA policies on behalf of the end user. For example, to require MFA for all activities:

```json
{
  "userId": "<suborg-root-user-id>",
  "mfaPolicyName": "MFA for everything",
  "condition": "true",
  "requiredAuthenticationMethods": /* The authentication methods you want to require */,
  "order": 1
}
```

## MFA recovery

> If an end user loses access to one of their authentication methods, they may be locked out of activities that require MFA. Because Turnkey cannot write to organizations directly, Turnkey is unable to recover users on their behalf. You must set up a recovery mechanism in advance.

Reviewer: ```suggestion
If an end user loses access to one of their authentication methods, they may be locked out of activities that require MFA. Because Turnkey cannot write to organizations directly, Turnkey is unable to recover users on their behalf. **You must set up a recovery mechanism in advance.**
```
